src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -434,12 +439,21 @@ private void CheckCaptured(Symbol variable, ParameterSymbol rangeVariableUnderly
         /// <param name="variable"></param>
         private void NoteCaptured(Symbol variable)
         {
-            if (variable.Kind != SymbolKind.RangeVariable || this.regionPlace == PreciseAbstractFlowPass<LocalState>.RegionPlace.Inside)
+            if (this.regionPlace == RegionPlace.Inside)
             {
+                _capturedInside.Add(variable);
+                _capturedVariables.Add(variable);
+            }
+            else if (variable.Kind != SymbolKind.RangeVariable)
+            {
+                _capturedOutside.Add(variable);
                 _capturedVariables.Add(variable);
             }
         }
 
+        protected IEnumerable<Symbol> GetCapturedInside() => _capturedInside.ToArray();
```

Comment:
Those two methods probably deserve the same comment as the following two methods ("do not expose poolable ...")